src/objects.c

```diff
@@ -2180,8 +2183,12 @@ static int match_public_keys(P11PROV_OBJ *key1, P11PROV_OBJ *key2)
     P11PROV_OBJ *priv_key;
     int ret = RET_OSSL_ERR;
 
-    if (key1->class == CKO_PUBLIC_KEY && key2->class == CKO_PUBLIC_KEY) {
-        /* both keys are public, match directly their public values */
+    if ((key1->class == CKO_PUBLIC_KEY && key2->class == CKO_PUBLIC_KEY) || key1->data.key.type == CKK_RSA) {
+        /* either keys are public, match directly their public values
+         * OR
+         * CKA_RSA keys (private/public) contain CKA_MODULUS / CKA_PUBLIC_EXPONENT
+         * - no need to find_associated_obj
+         */
         return cmp_public_key_values(key1, key2);
```

Comment:
you probably want to urn make check-style-fix to automatically reformat according to our rules.